### DISCUSSION TITLE
missing or unsuitable terminal: xterm-ghostty

### DISCUSSION THREAD

**Discussion by youkale** [ghostty-org/ghostty#4064] [Q&A] [answered]
OS: macOS 15.2
Chip: Apple m1 pro

step 1. 
     execute command:  tmux
step 2.
   command + T , open the tab 
   execute command:  ssh remote-server 
   when I connect the remote server, re-open the tmux session:
   $ tmux a 
   missing or unsuitable terminal: xterm-ghostty

-- 

remote-server:
OS: Ubuntu 24.04.1 LTS \n \l
env:
  XDG_SESSION_TYPE=tty
  TERM=xterm-ghostty
  SSH_TTY=/dev/pts/0

**Comment by feketegy** [comment, marked as answer]
This is answered in the docs here: https://ghostty.org/docs/help/terminfo


You would have to install either `xterm-ghostty` on the remote server OR define `SetEnv TERM=xterm-256color` in your local ssh config file, to not use `xterm-ghostty`.

**Comment by ifaakash** [comment]
I fixed this temporarily using below command:
```
TERM=xterm-256color tmux ls
```
NOTE: For any command that you want to run, TERM=xterm-256color will be attached as prefix

**Reply by danielwe** [reply]
Hi! Follow the link in the other comment for more up-to-date information about fixes/workarounds. A lot has happened in the 14 months since this discussion was active.

TL;DR: For the best experience, add `shell-integration-features = sudo,ssh-terminfo,ssh-env` to your Ghostty config.

**Comment by valankar** [comment]
I find it hard to believe that this does not work out of the box. The docs say "Ghostty is designed to work out of the box with no configuration for most users." My 2 cents is to just make this work for new users trying out ghostty to make the transition easier.

**Reply by rhodes-b** [reply]
you can search many previous discussions on why this is not a ghostty problem. you can read more from the link in the answer above.
https://ghostty.org/docs/help/terminfo

but a TLDR is that ghostty created its own terminfo instead of lying like many other terminals do (other terminals that didnt lie also had this problem Kitty being one). 
terminfo is a extremely outdated technology from like the 70s and 80s and its typically distributed through ncurses a entirely separate project from ghostty so there isn't much ghostty can do from its perspective.

ghostty was accepted into ncurses but it takes time for computers to have a new enough ncurses to have ghosttys entry by default thats a fatal flaw of ncurses in a few years it should be a non issue for the most part as most machines will have its entry

New users can very easily just make it work by using the configuration option danielwe mentioned above.

There is nothing that is going to change ghostty side as its not going to lie and pretend its another terminal by default, if that upsets you then are free to not use ghostty its a FOSS project

**Reply by danielwe** [reply]
> My 2 cents is to just make this work for new users trying out ghostty

@valankar I agree this would be wonderful! The reason we don't do it is that it requires shadowing an existing executable with a shell function. In other words, when typing `ssh`, you would not actually be executing `/usr/bin/ssh`, but instead a function injected into your shell by Ghostty. This is not OK---people need to be able to trust that a shell command is what it looks like (imagine if your terminal aliased `ls` to `rm -rf /` without your knowledge...). That's why we require explicit consent via a config option (`shell-integration-features = sudo,ssh-terminfo,ssh-env`) before injecting this function.

As @rhodes-b explained, this affects any new terminal, but works itself out over a couple of years as servers pick up the new terminfo entry through updates, so over time, fewer and fewer users will need to use the config option.

**Reply by 00-kat** [reply]
Locking; don't bump ancient discussions with such feedback: if you want to propose a way to make it work out of the box, make a new discussion.
